cmd/metal-api/internal/service/machine-service.go

```diff
@@ -1156,37 +1156,63 @@ func gatherNetworksFromSpec(ds *datastore.RethinkStore, allocationSpec *machineA
 		return nil, fmt.Errorf("given network ids are not unique")
 	}
 
-	if privateNetwork == nil {
+	if len(privateNetworks) == 0 {
 		return nil, fmt.Errorf("no private network given")
 	}
 
-	if privateNetwork.network.ProjectID != allocationSpec.ProjectID {
+	if !allocationSpec.IsFirewall && len(privateNetworks) > 1 {
+		return nil, fmt.Errorf("machines are not allowed to be placed into multiple private networks")
+	}
+
+	if len(privateNetworks) == 1 && privateNetworks[0].network.Shared {
+		// this means that this is a machine of a shared private network
+		// this is an exception where the primary private network is a shared one.
+		privateNetworks[0].isPrimaryPrivate = true
+		primaryPrivateNetwork = privateNetworks[0]
+	}
+
+	if primaryPrivateNetwork == nil {
+		return nil, fmt.Errorf("could not determine primary private network")
+	}
+
+	if primaryPrivateNetwork.network.ProjectID != allocationSpec.ProjectID {
 		return nil, fmt.Errorf("the given private network does not belong to the project, which is not allowed")
 	}
 
-	for _, ipString := range allocationSpec.IPs {
-		ip, err := ds.FindIPByID(ipString)
-		if err != nil {
-			return nil, err
-		}
-		if ip.ProjectID != allocationSpec.ProjectID {
-			return nil, fmt.Errorf("given ip %q with project id %q does not belong to the project of this allocation: %s", ip.IPAddress, ip.ProjectID, allocationSpec.ProjectID)
+	for _, pn := range privateNetworks {
+		if pn.network.PartitionID != partitionPrivateSuperNetwork.PartitionID {
+			return nil, fmt.Errorf("private networks must be in the partition where the machine is going to be placed")
 		}
-		network, ok := specNetworks[ip.NetworkID]
-		if !ok {
-			return nil, fmt.Errorf("given ip %q is not in any of the given networks, which is required", ip.IPAddress)
+
+		if !pn.isPrimaryPrivate && !pn.network.Shared {
+			return nil, fmt.Errorf("private network is not allowed to be used because it is not marked as shared")
 		}
-		s := ip.GetScope()
-		if s != metal.ScopeMachine && s != metal.ScopeProject {
-			return nil, fmt.Errorf("given ip %q is not available for direct attachment to machine because it is already in use", ip.IPAddress)
+
+		for _, ipString := range allocationSpec.IPs {
+			ip, err := ds.FindIPByID(ipString)
+			if err != nil {
+				return nil, err
+			}
+			if ip.ProjectID != allocationSpec.ProjectID {
```

Comment:
if the network is shared and the allocation contains a firewall, projectID is allowed to differ

Reply:
This condition is still valid: the ip address should have a reference to the project where it is used!

With auto acquisition this is the case: `makeMachineNetwork` will insert the projectID of the allocation.
Using a specific IP is also okay: `metalctl network ip allocate --network shared --project consumer`

Reply:
This was actually a bug. `metalctl network ip allocate` for network with differing project id should only succeed for shared networks.